### DISCUSSION TITLE
GraphQL Pages Tree Query

### DISCUSSION THREAD

**Discussion by esb** [requarks/wiki#5313] [Error / Bug Report]
Wiki.js 2.5.274

There are some problems with API queries using the tree query function.

1. The following query always fails:
  ```query { pages { tree(path: "domains/core/charm", mode: PAGES, locale: "en") { id path title parent }}}```
  The error returned is:
   ```builder.andWhereNotNull is not a function```
2. The schema for the tree function makes "locale" a required parameter. This does not appear to be necessary, as /wiki/server/graph/resolvers/page.js line 226 sets the local to the default value if none is provided.

**Comment by vuducmanh11** [comment]
I have the same problem, query pages tree with others mode(work properly)

**Comment by PythonCoderAS** [comment]
Ran into the same problem
